src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/PrereqsTests.cs

```diff
@@ -0,0 +1,249 @@
+/// </summary>
+/// <remarks>
+/// This class intentionally does not derive from the SmokeTests class.
+/// Because these tests need to be executed in separate test pass after all the other smoke tests have been executed, the test
+/// configuration makes use of Trait attributes so that the "--filter" option of "dotnet test" can be used to control which
+/// tests get executed for each test pass. This class defines a special "Prereqs" category whereas all the other smoke tests
+/// have a category of "Default" which is implemented by the base SmokeTests class. If this class were to derive from SmokeTests,
+/// it would cause these tests to be executed along with all the other smoke tests, which is not what we want.
+/// </remarks>
+[Trait("Category", "Prereqs")]
+public partial class PrereqsTests : IDisposable
+{
+    private static readonly Regex _dynamicVersionRegex = DynamicVersionRegex();
+    private static readonly Regex _dotNetSdkMajorMinorVersionRegex = DotNetSdkMajorMinorVersionRegex();
+
+    private DotNetHelper _dotNetHelper;
+    private ITestOutputHelper _outputHelper;
+    private string _tmpDir;
+
+    public PrereqsTests(ITestOutputHelper outputHelper)
+    {
+        _dotNetHelper = new DotNetHelper(outputHelper);
+        _outputHelper = outputHelper;
+        _tmpDir = Path.Combine(Directory.GetCurrentDirectory(), "prereqs-artifacts");
+        Directory.CreateDirectory(_tmpDir);
+    }
+
+    public void Dispose()
+    {
+        Directory.Delete(_tmpDir, recursive: true);
+    }
+
+    private static string GetSharedRuntimeVersion(string sharedRuntimeName)
+    {
+        string? version = null;
+        string path = $"./shared/{sharedRuntimeName}/";
+        SmokeTests.EnumerateTarball(Config.SdkTarballPath,
+            tarEntry =>
+            {
+                if (tarEntry.Name.StartsWith(path) && tarEntry.Name.Length > path.Length)
+                {
+                    version = Path.GetFileName(tarEntry.Name.TrimEnd('/'));
+                    return false;
+                }
+
+                return true;
+            });
+
+        Assert.NotNull(version);
+
+        return version!;
+    }
+
+    /// <summary>
+    /// Compares the restored packages from the smoke tests with the expected packages.
+    /// </summary>
+    [Fact]
+    public void ComparePackages()
+    {
+        string aspnetVersion = GetSharedRuntimeVersion("Microsoft.AspNetCore.App");
+        string runtimeVersion = GetSharedRuntimeVersion("Microsoft.NETCore.App");
+        string fsharpVersion = Config.FSharpPackageVersion;
+
+        string prereqsProjPath = Path.Combine(Directory.GetCurrentDirectory(), "assets", "prereqs.csproj");
+        XDocument doc = XDocument.Load(prereqsProjPath);
+
+        // Get all of the packages defined in prereqs project. There may be packages that have multiple versions
+        // listed.
+        IEnumerable<(string PackageName, IEnumerable<string> Versions)> expectedPackages = doc.Root!.Elements()
```

Comment:
This is certainly a valid option but I don't see it as providing much value over the existing implementation. It would just be a different way of implementing it. As for differences in complexity of the code, I see it as being a wash. In the interest of time and getting these changes in, I'm going to leave it as is.